app/gui/view/graph-editor/src/lib.rs

```diff
@@ -1744,16 +1751,110 @@ impl GraphEditorModel {
         //   `find_free_place` looks for free place for the origin point, and we want to fit not
         //   only the point, but the whole node.
         let node_areas = nodes.values().map(|node| {
-            let position = node.position();
-            let left = position.x - x_gap - min_spacing;
-            let right = position.x + node.view.model.width() + x_gap;
-            let top = position.y + node::HEIGHT + y_gap;
-            let bottom = position.y - node::HEIGHT - y_gap;
+            let bounding_box = node.frp.bounding_box.value();
+            let left = bounding_box.left() - x_gap - min_spacing;
+            let right = bounding_box.right() + x_gap;
+            let top = bounding_box.top() + node::HEIGHT / 2.0 + y_gap;
+            let bottom = bounding_box.bottom() - node::HEIGHT / 2.0 - y_gap;
             OccupiedArea { x1: left, x2: right, y1: top, y2: bottom }
         });
         find_free_place(starting_from, direction, node_areas)
     }
 
+    /// Calculate a position for a new node, starting from the mouse position and aligning to the
+    /// node at the source of the specified edge if the mouse position is close to the node.
```

Comment:
This sentence is hard to understand. What is "specified edge"? Do you mean the edge of the provided [`edge_id`]? Please divide that into shorter sentences or use more interpunction. Please apply to other comments as well This one should be:

Return a position for a newly created node, such that the node will not overlap with existing ones. The position is calculated by taking the mouse position and aligning it to the source node (the node at the source of the [`edge_id`] edge) if the source node is close to the mouse position. 

To learn more about the align algorithm, see the docs of [`new_node_position_aligned_if_close_to_node`].

Reply:
I modified the comment as suggested, thanks! However, after the refactoring proposed by Michael,  all the functions in the new module now return positions for nodes that will not overlap with existing nodes. As such, I moved the fragment mentioning that to the new module's docs to avoid repetition. However, if that was not a good idea, I'll bring this fragment back to the function documentation comments.